src/cluster_legacy.c

```diff
@@ -1902,10 +1902,10 @@ void clearNodeFailureIfNeeded(clusterNode *node) {
 
     serverAssert(nodeFailed(node));
 
-    /* For replicas we always clear the FAIL flag if we can contact the
-     * node again. */
+    /* For replicas or primaries without slots, we always clear the FAIL flag
+     * if we can contact the node again. */
     if (nodeIsReplica(node) || node->numslots == 0) {
```

Comment:
I'm struggling whether to modify this as well.

Reply:
I think it makes sense to replace this as well.